server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -990,19 +989,20 @@ static Settings aggregateIndexSettings(
 
             // Loop through all the explicit index setting providers, adding them to the
             // additionalIndexSettings map
+            final Settings.Builder additionalIndexSettings = Settings.builder();
             final var resolvedAt = Instant.ofEpochMilli(request.getNameResolvedAt());
             for (IndexSettingProvider provider : indexSettingProviders) {
-                additionalIndexSettings.put(
-                    provider.getAdditionalIndexSettings(
-                        request.index(),
-                        request.dataStreamName(),
-                        timeSeriesTemplate,
-                        currentState.getMetadata(),
-                        resolvedAt,
-                        templateAndRequestSettings,
-                        combinedTemplateMappings
-                    )
+                var newAdditionalSettings = provider.getAdditionalIndexSettings(
+                    request.index(),
+                    request.dataStreamName(),
+                    timeSeriesTemplate,
+                    currentState.getMetadata(),
+                    resolvedAt,
+                    templateAndRequestSettings,
+                    combinedTemplateMappings
                 );
+                validateAdditionalSettings(provider, newAdditionalSettings, additionalIndexSettings);
```

Comment:
Here we validate the ` newAdditionalSettings` against the `additionalSettings` seen so far and not values previously set by the user for example `templateAndRequestSettings`. Are these 2 settings sets disjoint ?

Reply:
The `newAdditionalSettings` variable contains any settings that the current index settings provider might have set. The `additionalIndexSettings` variable tracks any settings that any index settings provider may have set while looping over the list of index settings providers. The new validation requires that `newAdditionalSettings` cannot contain any settings that are already defined in  `additionalIndexSettings`.  So yes, these two Settings variables should be a disjoint.